service/contacts.js

```diff
@@ -1,27 +1,100 @@
+const { isValidObjectId } = require("mongoose");
 const Contact = require("./schemas/contacts");
 
-const getAllContacts = async () => Contact.find();
+const getAllContacts = async (owner) => {
+  try {
+    const contacts = await Contact.find({ owner });
+    return contacts;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const getContactById = async (contactId, userId) => {
+  if (!isValidObjectId(contactId)) {
+    throw new Error("Not found. Invalid Id!");
+  }
+  try {
+    const selectedConctact = await Contact.findOne({ _id: contactId });
+    const contactOwner = selectedConctact.owner;
+
+    if (userId.toString() !== contactOwner.toString()) {
+      throw new Error("Not authorized");
+    }
+    return selectedConctact;
+  } catch (error) {
+    throw error;
+  }
+};
+
+const updateContact = async (contactId, body, userId) => {
+  if (!isValidObjectId(contactId)) {
+    throw new Error("Not found. Invalid Id!");
+  }
+  try {
+    const selectedContact = await Contact.findOne({ _id: contactId });
+    const contactOwner = selectedContact.owner;
 
-const getContactById = async (contactId) => Contact.findById(contactId);
+    if (userId.toString() !== contactOwner.toString()) {
+      throw new Error("Not authorized.");
+    }
 
-const createContact = async ({ name, email, phone, favorite }) => {
-  return Contact.create({ name, email, phone, favorite });
+    await Contact.findByIdAndDelete(contactId, body);
```

Comment:
A sprawdzałeś czy nie da się wysłać więcej filtrów? Bo z tego co kojarzę to można dodać tez obiekt też.

Reply:
Nie rozumiem...?

Reply:
coś takiego: 
```
db.getCollection("movies").find({genres: "Short", cast: "A.C. Abadie"})
```
Są dwa filtry, można zrobić _id i owner. I wtedy baza danych Ci sprawdzi czy taki rekord istnieje. Nie musisz tego sama sprawdzać potem.

Reply:
Tylko, że ja już to wcześniej zmieniłam.... chyba znów nie wszedł mi push zanim sprawdziłeś, dlatego nie wiedizałam o co chodzi:(